components/server/src/api/auth-provider-service-api.ts

```diff
@@ -118,9 +117,13 @@ export class AuthProviderServiceAPI implements ServiceImpl<typeof AuthProviderSe
     @Unauthenticated()
     async listAuthProviderDescriptions(
         request: ListAuthProviderDescriptionsRequest,
-        context: HandlerContext,
+        _: HandlerContext,
     ): Promise<ListAuthProviderDescriptionsResponse> {
-        const user = context.user;
+        const userId = ctxUserId();
```

Comment:
@geropl, just realized, this actually throws if unauthenticated.

Need to address it, as it breaks Login page with gRPC enabled.

Reply:
https://github.com/gitpod-io/gitpod/pull/19094

Reply:
@AlexTugarev Yes, this one throws. If we need one that does not throw, we should name it `ctxTryUserId`, following the pattern in request-context.ts :+1: